### PR TITLE
Implemented new Amazon.Lambda.MQEvents package.

### PR DESCRIPTION
**Pull request by ashishdhingra**
*Issue #, if available:* #1048

*Description of changes:*
Implemented new `Amazon.Lambda.MQEvents` package.

___
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Libraries/src/Amazon.Lambda.AmazonMQEvents/Amazon.Lambda.AmazonMQEvents.csproj [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <Import Project="..\..\..\buildtools\common.props" />
+	
+  <PropertyGroup>
+    <TargetFrameworks>netstandard2.0;netcoreapp3.1</TargetFrameworks>
+    <Description>Amazon Lambda .NET Core support - AmazonMQEvents package.</Description>
+    <AssemblyTitle>Amazon.Lambda.AmazonMQEvents</AssemblyTitle>
+    <VersionPrefix>1.0.0</VersionPrefix>
+    <AssemblyName>Amazon.Lambda.AmazonMQEvents</AssemblyName>
```
Comment (ganeshnj) [inline Libraries/src/Amazon.Lambda.AmazonMQEvents/Amazon.Lambda.AmazonMQEvents.csproj:10]:
```suggestion
    <AssemblyName>Amazon.Lambda.MQEvents</AssemblyName>
```

I don't see yet another Amazon prefix before event names, this should just be `MQEevents`.

File: Libraries/test/EventsTests.Shared/EventTests.cs [hunk 2]
```diff
@@ -1933,6 +1933,87 @@ private void Handle(KafkaEvent kafkaEvent)
             }
         }
 
+        [Theory]
```
Comment (ganeshnj) [inline Libraries/test/EventsTests.Shared/EventTests.cs:1936]: Is this change integration tested?

File: Libraries/src/Amazon.Lambda.AmazonMQEvents/ActiveMQEvent.cs [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        /// <summary>
+        /// List of ActiveMQ messages.
+        /// </summary>
+        public IList<ActiveMQMessage> Messages { get; set; }
+
+        /// <summary>
+        /// Kafka event record.
+        /// </summary>
+        public class ActiveMQMessage
+        {
```
Comment (ganeshnj) [inline Libraries/src/Amazon.Lambda.AmazonMQEvents/ActiveMQEvent.cs:31]: Missed?
- deliveryMode
- correlationID
- replyTo
- type
- expiration
- priority

FYI 
https://github.com/aws/aws-lambda-java-libs/blob/311f97cc47b414633c62ee96d9df3e8404e14d69/aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/ActiveMQEvent.java#L48

**Review by ganeshnj — CHANGES_REQUESTED** [review]

File: Libraries/src/Amazon.Lambda.AmazonMQEvents/Amazon.Lambda.AmazonMQEvents.csproj (hunk 1 above)
Comment (ashishdhingra) [reply Libraries/src/Amazon.Lambda.AmazonMQEvents/Amazon.Lambda.AmazonMQEvents.csproj:10]: The reason I prefixed it with `Amazon` is that the documentation page https://aws.amazon.com/amazon-mq/?amazon-mq.sort-by=item.additionalFields.postDateTime&amazon-mq.sort-order=desc mentions it as `Amazon MQ`. Please advise if we should still rename it to `MQEvents`. If yes, I would need to submit new revision of pipeline CR to update it with new name.

File: Libraries/src/Amazon.Lambda.AmazonMQEvents/ActiveMQEvent.cs (hunk 3 above)
Comment (ashishdhingra) [reply Libraries/src/Amazon.Lambda.AmazonMQEvents/ActiveMQEvent.cs:31]: As I figured it out earlier, the above Java library appears to be out of date. Please refer https://docs.aws.amazon.com/lambda/latest/dg/with-mq.html for the event structure which was used as reference to develop this POCO class.

File: Libraries/test/EventsTests.Shared/EventTests.cs (hunk 2 above)
Comment (ashishdhingra) [reply Libraries/test/EventsTests.Shared/EventTests.cs:1936]: Could you please elaborate? I didn't setup ActiveMQ or RabbitMQ cluster and configured it with Lambda. I simply used guidance from sample Lambda events at https://docs.aws.amazon.com/lambda/latest/dg/with-mq.html.

File: Libraries/src/Amazon.Lambda.AmazonMQEvents/Amazon.Lambda.AmazonMQEvents.csproj (hunk 1 above)
Comment (ashishdhingra) [reply Libraries/src/Amazon.Lambda.AmazonMQEvents/Amazon.Lambda.AmazonMQEvents.csproj]: @ganeshnj I have renamed the package to `Amazon.Lambda.MQEvents`. Please review.

Comment (ganeshnj) [reply Libraries/src/Amazon.Lambda.AmazonMQEvents/Amazon.Lambda.AmazonMQEvents.csproj]: Awesome thanks for making the change.

File: Libraries/src/Amazon.Lambda.AmazonMQEvents/ActiveMQEvent.cs (hunk 3 above)
Comment (ganeshnj) [reply Libraries/src/Amazon.Lambda.AmazonMQEvents/ActiveMQEvent.cs]: If Java library is not up to date then could you cut a ticket to Java team to update their package?

Comment (ashishdhingra) [reply Libraries/src/Amazon.Lambda.AmazonMQEvents/ActiveMQEvent.cs]: I have created internal ticket for AWS SDK for Java.

File: Libraries/test/EventsTests.Shared/EventTests.cs (hunk 2 above)
Comment (ashishdhingra) [reply Libraries/test/EventsTests.Shared/EventTests.cs:1936]: Tested end-to-end using instructions at https://awscloudfeed.com/whats-new/compute/using-amazon-mq-as-an-event-source-for-aws-lambda (reused the same instructions for Active MQ and Rabbit MQ).

**NOTE:** For RabbitMQ, there was intermittent error where it was unable to deserialize priority. The error went away after creating a test Node.js Lambda function to examine the event JSON (it reported `partition` as integer value). After some testing, the error in .NET Lambda went away.

Comment (ganeshnj) [reply Libraries/test/EventsTests.Shared/EventTests.cs:1936]: I don't even see `partition` in the event models. has it been missed?

Comment (ashishdhingra) [reply Libraries/test/EventsTests.Shared/EventTests.cs:1936]: Sorry, it's `priority`. I was working on multiple items in parallel.

**Review by ganeshnj — APPROVED** [review]

File: Libraries/src/Amazon.Lambda.MQEvents/ActiveMQEvent.cs [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+            /// <summary>
+            /// Message Destination
+            /// </summary>
+            public Destination Destination { get; set; }
+
+            /// <summary>
+            /// Message Timestamp
+            /// </summary>
+            public long? Timestamp { get; set; }
```
Comment (ashovlin) [inline Libraries/src/Amazon.Lambda.MQEvents/ActiveMQEvent.cs:70]: I haven't worked with our event packages a lot - do we ever convert these to `DateTime` here? Or does the user have to do that?

Comment (ashishdhingra) [reply Libraries/src/Amazon.Lambda.MQEvents/ActiveMQEvent.cs:70]: Ideally that should be the case. However, I have seen instances where converting epoch time to DateTime gives an error in our Lambda library (there is already an open issue https://github.com/aws/aws-lambda-dotnet/issues/839). Also, this was ported from https://github.com/aws/aws-lambda-java-libs/blob/master/aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/ActiveMQEvent.java, which uses `long`. So wanted to be consistent here.

File: Libraries/src/Amazon.Lambda.MQEvents/ActiveMQEvent.cs [hunk 5]
```diff
@@ -0,0 +1,94 @@
+﻿namespace Amazon.Lambda.MQEvents
```
[the remark this answers is not in the document]
Comment (ashishdhingra) [reply Libraries/src/Amazon.Lambda.MQEvents/ActiveMQEvent.cs:1]: Skipping it for now based on discussion. We should plan to add it in every events package seperately.

File: Libraries/src/Amazon.Lambda.MQEvents/ActiveMQEvent.cs (hunk 4 above)
Comment (ashovlin) [reply Libraries/src/Amazon.Lambda.MQEvents/ActiveMQEvent.cs:70]: Sounds good, thanks for the context.

**Review by ashovlin — APPROVED** [review]

**Pull request merged**